pallets/process-validation/src/restrictions.rs

```diff
@@ -77,6 +84,17 @@ where
             let meta = selected_output.metadata.get(&metadata_key);
             meta == Some(&metadata_value)
         }
+        Restriction::SenderHasInputRole { index, role_key } => {
+            let selected_input = &inputs[index as usize];
+            let sender_has_role = match selected_input.roles.get(&role_key) {
+                Some(account) => sender == account,
+                None => false,
+            };
+            if !sender_has_role {
+                return false;
+            }
+            true
```

Comment:
this can be simplified I think to:

```rust
match selected_input.roles.get(&role_key) {
   Some(account) => sender == account,
   None => false,
 }
```

Reply:
done